### PR TITLE
replace static with inline, and some cosmetic fixes

### PR DESCRIPTION
**Pull request by poul250**

### CODE REVIEW

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-#define PLATFORM_DELEGATES_CREATEDELEGATE
+#ifndef PLATFORM_DELEGATES_CREATEDELEGATE_H
+#define PLATFORM_DELEGATES_CREATEDELEGATE_H
 
 #include "Platform.Delegates.Delegate.h"
 
 namespace Platform::Delegates
 {
     template <typename ReturnType, typename... Args>
-    static Delegate<ReturnType(Args...)> CreateDelegate(ReturnType(&function)(Args...))
```
Comment (Konard) [inline cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:11]: Why did you remove static keyword?

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h [hunk 2]
```diff
@@ -1,29 +1,27 @@
-#pragma once
```
Comment (Konard) [inline cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:1]: I hope that helps 👍

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-    static Delegate<ReturnType(Args...)> CreateDelegate(ReturnType(&function)(Args...))
+    inline Delegate<ReturnType(Args...)> CreateDelegate(ReturnType(&function)(Args...))
     {
-        return Delegate<ReturnType(Args...)>(function);
+        return Delegate<ReturnType(Args...)>{function};
     }
 
     template <typename ReturnType, typename... Args>
-    static Delegate<ReturnType(Args...)> CreateDelegate(const std::function<ReturnType(Args...)> &function)
+    inline Delegate<ReturnType(Args...)> CreateDelegate(const std::function<ReturnType(Args...)> &function)
```
Comment (Konard) [inline cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:15]: Doesn`t C++ compiler inline methods by default?

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h (hunk 1 above)
Comment (poul250) [reply cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:11]: static specifier means internal linkage, so every compilation unit will have his own version of this functions.

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h (hunk 3 above)
Comment (poul250) [reply cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:15]: it doesn't. C++ compiler can [not] inline functions itself, and 'inline' specifier is just a recommendation. 'Inline' keyword is used here for "weak external" linkage, thanks to which multiple declaration does not violate One Definition Rule and it links only one function in all translation units.

Comment (Konard) [reply cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:15]: Do we really need explicit `inline` keyword here? Will the compiler do implicit inlining here without the keyword?

Comment (Konard) [reply cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:15]: https://en.wikipedia.org/wiki/One_Definition_Rule ... well, ok, lets try this.

File: cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h (hunk 1 above)
Comment (Konard) [reply cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h:11]: Good to know. Looks like `static` keyword have different meanings in different contexts.
